nix/build.nix

```diff
@@ -0,0 +1,153 @@
+# Building Limine with all features in Nix.
+#
+# Independent of the packing in nixpkgs, This is convenient for prototyping and
+# local development.
+#
+# These derivations always builds Limine from the current src tree and not some
+# stable release. Further, unlike the nixpkgs derivation, this derivation runs
+# the ./bootstrap step which needs network access. Due to the nature of the
+# self-hacked Git submodules download approach, packaging this project in Nix
+# is especailly complicated. The complicated multi-derivation approach below
+# is the best I can get after multiple hours of trying. :).
+
+{
+  # Helpers
+  fd
+, lib
+, keep-directory-diff
+, nix-gitignore
+, stdenv
+, stdenvNoCC
+
+  # Actual derivation dependencies.
+, autoconf
+, automake
+, cacert
+, git
+, gnumake
+, llvmPackages
+, mtools
+, nasm
+, pkg-config
+}:
+
+let
+  # The actual current source but close to the repo state, i.e., no files from
+  # the bootstrap step.
+  currentRepoSrc = nix-gitignore.gitignoreSource [
+    # Prevent that the hash ov the sources change, when we change it in Nix.
+    "flake.nix"
+    "flake.lock"
+    "nix/"
+
+    # bootstrapped sources from ./bootstrap
+    "/build-aux/freestanding-toolchain"
+    "/freestanding-headers"
+    "/decompressor/cc-runtime"
+    "/limine-efi"
+    "/tinf"
+    "/common/flanterm"
+    "/stb"
```

Comment:
Similar to #342, shouldn't this include *all* files and directories as shown by `git status --ignored` after `./bootstrap`?

Reply:
Yikes. My mistake. As they are already excluded by `.gitignore`, `nix-gitignore.gitignoreSource` already takes care of that. The list are only additional entries.